src/components/Notifications/EventLog/EventLogTreeFilter.tsx

```diff
@@ -0,0 +1,211 @@
+        init[group.name] = {
+            id: group.name,
+            name: group.displayName,
+            checkProps: { checked: checkAll || (!currentFilter ? false : null) },
+            children: items.length !== 0 ? items.map(item => ({
+                id: item.name,
+                name: item.displayName,
+                checkProps: { checked: checkAll || currentFilterChipValues?.includes(item.name) }
+            })) : undefined
+        };
+    });
+
+    return init;
+};
+
+export const EventLogTreeFilter: React.FunctionComponent<EventLogTreeFilterProps> = (props) => {
+    const { groups, placeholder, filters, updateFilters } = props;
+
+    const initialize = React.useMemo(() => initTreeNodeById(groups, filters), [ groups, filters ]);
+
+    const [ treeNodeById, setTreeNodeById ] = React.useState<TreeNodeDict>(initialize);
+    const [ isToggled, setIsToggled ] = React.useState(false);
+
+    const treeDataArray = React.useMemo(() => treeNodeById ? Object.values(treeNodeById) : [], [ treeNodeById ]);
+
+    React.useEffect(() => {
+        if (groups.length !== 0) {
+            setTreeNodeById(produce((prev) => {
+                if (Object.keys(prev).length === 0) {
+                    return initialize;
+                } else if (filters.length === 0) {
+                    return initialize;
+                } else {
+                    filters.forEach(activeFilter => {
+                        const treeNode = prev[activeFilter.bundleId];
+                        const activeChips = activeFilter.chips.map(chip => chip.value);
+
+                        treeNode.children?.forEach(childNode => {
+                            childNode.checkProps.checked = activeChips.includes(childNode.id);
+                        });
+
+                        if (allChildrenChecked(treeNode)) {
+                            treeNode.checkProps.checked = true;
+                        } else if (childChecked(treeNode)) {
+                            treeNode.checkProps.checked = null;
+                        } else {
+                            treeNode.checkProps.checked = false;
+                        }
+                    });
+                }
+            }));
+        }
+    }, [ groups, filters, initialize ]);
+
+    const flattenTree = React.useMemo(() => {
+        const flatTreeDataArray = produce(treeDataArray, (prev) => {
+            prev.forEach(treeNode => {
+                if (treeNode.children) {
+                    prev.push(...treeNode.children);
+                }
+            });
+        });
+
+        return flatTreeDataArray;
+    }, [ treeDataArray ]);
+
+    React.useEffect(() => {
+        const groupNames = groups.map(group => group.name);
+        const activeParentFilters = flattenTree.filter(treeNode => {
+            const isActive = treeNode.checkProps.checked;
+            return (isActive || isActive === null) && groupNames.includes(treeNode.id);
```

Comment:
What's the format of id? Is it like `the-group-id.the-child-id` ? If so maybe we can change this to something like:
```suggestion
            return (isActive || isActive === null) && treeNode.id.startsWith(`${groupNames}.`);
```

to avoid picking up false positives from the include in case the app name is similar to a bundle.

Reply:
groupId.childId format is only used for the URL query param... the treeNodes carry just the id of whatever thing it is representing, so app would be appId... bundle would be bundleId

Reply:
So it is possible to have an application with the same name as a bundle?

Reply:
We don't have that now, but not imposible to have some sort of clash in the future think of a group called: "application" and an app called "application-services". We don't enforce what an app or a bundle can be called so we can't make the assumption that it won't happen.

Reply:
groupNames is an array, it would only return true if any instance in the string array matches exactly with the instance of the treeNode id.... so they would have to be an exact match in terms of names. I think there are other areas that would need some major refactoring if an assumption was that app names and bundle names could be the same.

Reply:
ah - nevermind my comment then :) for a moment thought that `groupNames` was a string.